tests/valgrind/potential-memleaks.c

```diff
@@ -0,0 +1,38 @@
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include <openssl/aes.h>
+
+#define KEYLEN 128
+
+/* Keep all code in `int main` to simplify the valgrind suppression. */
+int
+main()
+{
+
+	/* Problem with FreeBSD fgets() with stdin. */
+	{
```

Comment:
I'm guessing that you won't like these nested blocks?  I'd like to keep variables and code for each "potential memleak" together, but putting them into a function makes generating a suitable suppression file a bit more annoying (it takes 3 `grep` commands instead of 2).